org.eclipse.xtext.xtext.ui.examples/projects/arithmetics/org.eclipse.xtext.example.arithmetics.ui/src/org/eclipse/xtext/example/arithmetics/ui/autoedit/InterpreterAutoEdit.xtend

```diff
@@ -18,14 +18,13 @@ import org.eclipse.xtext.example.arithmetics.interpreter.Calculator
 import org.eclipse.xtext.nodemodel.util.NodeModelUtils
 import org.eclipse.xtext.resource.XtextResource
 import org.eclipse.xtext.ui.editor.model.XtextDocumentUtil
-import com.google.inject.Inject
 
 /** 
  * An interactive interpreter as an {@link IAutoEditStrategy}
  */
 class InterpreterAutoEdit implements IAutoEditStrategy {
 	
-	@Inject extension XtextDocumentUtil
+	extension XtextDocumentUtil = new XtextDocumentUtil
```

Comment:
did you have a look how we create strategies in general. it differs from the pattern here

Reply:
Yes, it is different. Usually we use an injected provider.

Reply:
@Inject
	protected Provider<DefaultIndentLineAutoEditStrategy> defaultIndentLineAutoEditStrategy;
	

in the superclass and not new as in the the example
acceptor.accept(new InterpreterAutoEdit(), IDocument.DEFAULT_CONTENT_TYPE)